forch/simple_auth_state_machine.py

```diff
@@ -95,28 +103,46 @@ def received_radius_reject(self):
                 self._logger.warning(
                     'Unexpected RADIUS response for %s, Ignoring it.', self.src_mac)
                 return
-            self._state_transition(self.UNAUTH, self.REQUEST)
-            self._current_timeout = time.time() + self._rej_timeout_sec
-            self._retry_backoff = 0
+            self._reset_state_machine()
             self._auth_callback(self.src_mac, self.UNAUTH, None, None)
 
     def handle_sm_timer(self):
-        """Handle timer timeout and check.trigger timeout behavior of states"""
+        """
+        Handle timer timeout and check.trigger timeout behavior of states:
+        * REQUEST state --request timeout & backoff < max_backoff--> REQUEST state + send request
```

Comment:
This syntax is very confusing. what's "check.trigger"? What's the -- before request mean?   Please try to clean it up  a bit so it's easier to understand.  "state" is redundant with ALLCAPS names.

Reply:
updated